src/components/v5/common/ActionSidebar/partials/VerifiedMembersSelect/VerifiedMembersSelect.tsx

```diff
@@ -0,0 +1,173 @@
+import { SealCheck } from '@phosphor-icons/react';
+import clsx from 'clsx';
+import { utils } from 'ethers';
+import React, { type FC } from 'react';
+import { useController, useFormContext } from 'react-hook-form';
+
+import { useAdditionalFormOptionsContext } from '~context/AdditionalFormOptionsContext/AdditionalFormOptionsContext.tsx';
+import { useMobile } from '~hooks';
+import useRelativePortalElement from '~hooks/useRelativePortalElement.ts';
+import useToggle from '~hooks/useToggle/index.ts';
+import { formatText } from '~utils/intl.ts';
+import Button from '~v5/shared/Button/Button.tsx';
+import SearchSelect from '~v5/shared/SearchSelect/index.ts';
+import UserAvatar from '~v5/shared/UserAvatar/index.ts';
+
+import { useVerifiedMembersSelect } from './hooks.ts';
+import { type VerifiedMembersSelectProps } from './types.ts';
+
+const displayName = 'v5.common.ActionsContent.partials.VerifiedMembersSelect';
+
+const VerifiedMembersSelect: FC<VerifiedMembersSelectProps> = ({
+  name,
+  fieldArrayMethods,
+}) => {
+  const {
+    field,
+    fieldState: { error },
+  } = useController({
+    name,
+  });
+  const isError = !!error;
+  const { usersOptions } = useVerifiedMembersSelect();
+  const [
+    isUserSelectVisible,
+    {
+      toggle: toggleUserSelect,
+      toggleOff: toggleOffUserSelect,
+      registerContainerRef,
+    },
+  ] = useToggle();
+  const { readonly } = useAdditionalFormOptionsContext();
+  const isMobile = useMobile();
+  const selectedMember = usersOptions.options.find((user) => {
+    return user.value === field.value.value;
+  });
+
+  const { portalElementRef, relativeElementRef } = useRelativePortalElement<
+    HTMLButtonElement,
+    HTMLDivElement
+  >([isUserSelectVisible], {
+    withMaxHeight: false,
+    withAutoTopPlacement: true,
+    withLeftPosition: !isMobile,
+  });
+  const { getValues, setValue } = useFormContext();
+  const checkedItems = getValues()
+    .members.filter((obj) => typeof obj.value === 'string')
+    .map((obj) => obj.value);
+
+  const onMemberSelect = (value: string | undefined) => {
+    if (!value || !utils.isHexString(value)) {
+      return;
+    }
+
+    const formValues = getValues().members;
+    const valueIndex = formValues.findIndex((item) => item?.value === value);
+
+    if (valueIndex !== -1) {
+      fieldArrayMethods.remove(valueIndex);
+      return;
+    }
+
+    const firstEmptyValueIndex = formValues.findIndex((item) => !item?.value);
+
+    if (formValues.length === 0 || firstEmptyValueIndex !== -1) {
+      const fieldName =
+        firstEmptyValueIndex === -1
+          ? 'members'
+          : `members.${firstEmptyValueIndex}`;
+      setValue(fieldName, { value });
+      return;
+    }
+
+    fieldArrayMethods.append({ value });
+  };
+
+  return (
+    <div className="sm:relative w-full flex items-center">
+      {readonly ? (
+        <>
+          <UserAvatar
+            user={selectedMember || field.value.value}
```

Comment:
could we save `field.value.value` to a variable called `walletAddress` and use that everywhere?